client/dashboard/sites/site-delete-modal/index.tsx

```diff
@@ -0,0 +1,247 @@
+			return __(
+				"We are still in the process of removing your previous plan. Please check back in a few minutes and you'll be able to delete your site."
+			);
+		}
+
+		if ( p2HubP2Count ) {
+			return sprintf(
+				/* translators: %d is the number of P2 in your workspace */
+				_n(
+					'There is %d P2 in your workspace. Please delete it prior to deleting your workspace.',
+					'There are %d P2s in your workspace. Please delete them prior to deleting your workspace.',
+					p2HubP2Count
+				),
+				p2HubP2Count
+			);
+		}
+
+		if ( isTrialSite ) {
+			return __(
+				'You have an active or expired free trial on your site. Please cancel this plan prior to deleting your site.'
+			);
+		}
+
+		return __(
+			'You have active paid upgrades on your site. Please cancel your upgrades prior to deleting your site.'
+		);
+	};
+
+	const renderPrimaryButton = () => {
+		if ( isAtomicRemovalInProgress ) {
+			return (
+				<Button variant="primary" onClick={ onClose }>
+					{ __( 'OK' ) }
+				</Button>
+			);
+		}
+
+		if ( p2HubP2Count ) {
+			return (
+				<Button variant="primary" href="/sites">
+					{ __( 'Manage sites' ) }
+				</Button>
+			);
+		}
+
+		if ( isTrialSite ) {
+			<Button variant="primary" href={ `/purchases/subscriptions/${ site.slug }` }>
+				{ __( 'Cancel trial' ) }
+			</Button>;
+		}
+
+		return (
+			<Button variant="primary" href={ `/purchases/subscriptions/${ site.slug }` }>
+				{ __( 'Manage purchases' ) }
+			</Button>
+		);
+	};
+
+	return (
+		<Modal title={ __( 'Unable to delete site' ) } size="medium" onRequestClose={ onClose }>
+			<VStack spacing={ 4 }>
+				<Text as="p">{ renderWarningContent() }</Text>
+				<HStack justify="flex-end">
+					{ ! isAtomicRemovalInProgress && (
+						<Button variant="tertiary" onClick={ onClose }>
+							{ __( 'Cancel' ) }
+						</Button>
+					) }
+					{ renderPrimaryButton() }
+				</HStack>
+			</VStack>
+		</Modal>
+	);
+}
+
+export function SiteDeleteConfirmModal( { site, onClose }: { site: Site; onClose: () => void } ) {
+	const router = useRouter();
+	const { createSuccessNotice, createErrorNotice } = useDispatch( noticesStore );
+	const [ formData, setFormData ] = useState< SiteDeleteFormData >( { domain: '' } );
+	const mutation = useMutation( deleteSiteMutation( site.slug ) );
+
+	const fields: Field< SiteDeleteFormData >[] = [
+		{
+			id: 'domain',
+			label: __( 'Type the site domain to confirm' ),
+			type: 'text' as const,
+			description: sprintf(
+				/* translators: %s: site domain */
+				__( 'The site domain is: %s' ),
+				site.slug
+			),
+		},
+	];
+
+	const form = {
+		type: 'regular' as const,
+		fields: [ 'domain' ],
+	};
+
+	const handleSubmit = ( e: React.FormEvent ) => {
+		e.preventDefault();
+
+		mutation.mutate( undefined, {
+			onSuccess: () => {
+				router.navigate( { to: '/sites' } );
+				createSuccessNotice(
+					sprintf(
+						/* translators: %s: site name */
+						__( '%s has been deleted.' ),
+						site.name
```

Comment:
nit: I think it's better to use slug instead of name, since in the modal we display the domain as well.

![Screenshot 2025-05-29 at 11 34 32 AM](https://github.com/user-attachments/assets/5c8081e5-4e07-458c-886f-316e36680ee2)

Reply:
I noticed Figma uses the site.name in the snackbar notice. I can change back to the site.slug 🙂